src/schemas/server.ts

```diff
@@ -177,37 +245,156 @@ export class Server {
-        const shoutout: Shoutout = (<ServerT> this.data).shoutout;
+        const shoutout: Shoutout = this.data.shoutout;
         return shoutout.channel && shoutout.role ? shoutout : null;
     }
 
     public set shoutout(data: Shoutout) {
-        (<ServerT> this.data).shoutout = data;
+        this.data.shoutout = data;
     }
 
     /**
      * The auto go-live post data
      */
     public get goLive(): GoLive {
-        const goLive: GoLive = (<ServerT> this.data).goLive;
+        const goLive: GoLive = this.data.goLive;
         return goLive.channel ? goLive : null;
     }
 
     public set goLive(data: GoLive) {
-        (<ServerT> this.data).goLive = data;
+        this.data.goLive = data;
+    }
+
+    /**
+     * All reaction roles of a server
+     * 
+     * @readonly
+     */
+    public get reactionRoles(): ReadonlyMap<Snowflake, ReactionRoles> {
+        return this.data.reactionMessages;
+    }
+
+    /**
+     * Retrieve the reaction role ID associated with a certain message ID
+     * and emoji ID. Returns null if there is no reaction role associated.
+     * 
+     * @param message the reaction role message ID
+     * @param emoji the reaction role emoji ID
+     * @returns the reaction role mode and role ID, or null if not a valid reaction role
+     */
+    public getReactionRoles(
+        message: Snowflake,
+    ): ReactionRoles {
+        // Get the reaction roles for the message
+        const roles: ReactionRoles = this.data.reactionMessages
+            ?.get(message);
+
+        // Return null if the message doesn't have reaction roles
+        if (!roles) {
+            return null;
+        }
+
+        // Get the reaction role connected to the emoji
+        return roles;
+    }
+
+    /**
+     * Sets reaction roles for a message. This cannot be changed later, only deleted.
+     * 
+     * @param message the reaction role channel ID
+     * @param message the reaction role message ID
+     * @param mode the reaction role mode
+     * @param reactionRoles the collection of reaction roles
+     */
+    public setReactionRoles(
+        channel: Snowflake,
+        message: Snowflake,
+        mode: ReactionRoleStyle,
+        reactionRoles: Map<Snowflake, Snowflake>
+    ): void {
+        // Get the collection of reaction role messages
+        let messages: ReactionMessages = this.data.reactionMessages;
+
+        // Create a new collection if none was found
+        if (!messages) {
+            this.data.reactionMessages = new Map<Snowflake, ReactionRoles>();
+            messages = this.data.reactionMessages;
+        }
+
+        // Set a new reaction role message
+        const roles: ReactionRoles = {
+            channel: channel,
+            mode: mode,
+            reactionRoles: reactionRoles
+        };
+
+        messages.set(message, roles);
+    }
+
+    public deleteReactionRoles(message: Snowflake): void {
+        this.data.reactionMessages.delete(message);
     }
```

Comment:
The reaction roles have their own setters and getters along with a getter that retrieves all reaction roles, and a delete method for removing reaction roles.